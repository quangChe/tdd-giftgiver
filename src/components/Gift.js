import React from 'react';

class Gift extends React.Component {
  state = {
    person: '',
    present: ''
  }

  changePerson = (e) => {
    const person = e.target.value;
    this.setState({ person });
  }

  render() {
    return (
      <div>
        <input
          type='text'
          onChange={this.changePerson}
          className='input-person' />
      </div>
    )
  }
}

export default Gift;