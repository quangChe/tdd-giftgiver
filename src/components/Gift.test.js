import React from 'react';
import { shallow } from 'enzyme';
import Gift from './Gift';

describe('Gift', () => {
  const gift = shallow(<Gift />);

  it('renders without crashing', () => {
    expect(gift).toMatchSnapshot();
  })

  it('initializes a person and present in `state`', () => {
    expect(gift.state()).toEqual({person: '', present: ''});
  })

  describe('when typing into the person input', () => {
    const INPUT_VALUE = 'Uncle Burt';

    beforeEach(() => {
      gift.find('.input-person').simulate('change', { target: { value: INPUT_VALUE } });
    })

    it('updates the person in `state`', () => {
      expect(gift.state().person).toEqual(INPUT_VALUE);
    })
  })
})