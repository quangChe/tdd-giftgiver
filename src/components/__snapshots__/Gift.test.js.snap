// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift renders without crashing 1`] = `
<div>
  <input
    className="input-person"
    onChange={[Function]}
    type="text"
  />
</div>
`;
